forc-pkg/src/source/mod.rs

```diff
@@ -179,19 +179,19 @@ impl Source {
 
     /// If a patch exists for this dependency source within the given project
     /// manifest, this returns the patch.
-    fn dep_patch<'manifest>(
+    fn dep_patch(
         &self,
         dep_name: &str,
-        manifest: &'manifest PackageManifestFile,
-    ) -> Option<&'manifest manifest::Dependency> {
+        manifest: &PackageManifestFile,
+    ) -> Result<Option<manifest::Dependency>> {
         if let Source::Git(git) = self {
-            if let Some(patches) = manifest.patch(&git.repo.to_string()) {
+            if let Some(patches) = manifest.resolve_patch(&git.repo.to_string())? {
```

Comment:
Internally, this calls calls the `PackageManifest::workspace` method which does the I/O work of finding the workspace, reading the manifest then deserializing it every time we call this `resolve_patch` method.

I think this approach is fine for now, though in a follow-up PR we may want to consider refactoring this so that we find, read and deserialize the resolved patch table only *once*, as if the file is edited during the compilation process (not uncommon while running sway-lsp) we might occasionally encounter some strange inconsistencies. E.g. it could be worth doing this upon loading the manifest file itself, or immediately after. Then, our `PatchMap` type (or similar abstraction) could be the type that gets passed in here to check for patches.